obspy/signal/cross_correlation.py

```diff
@@ -849,6 +850,123 @@ def templates_max_similarity(st, time, streams_templates):
         return 0
 
 
+def xcorr_detector(stream, template, mean_threshold=None, condition=None,
+                   template_time=None, holdon=None, holdoff=None,
+                   plot_ccmean=False, **kwargs):
+    """
+    Detector based on the cross correlation of a template with data.
+
+    :param stream: Stream with data traces
+    :param template: Stream with template traces (should be shorter than data)
+    :param mean_thresshold: mean cross correlation value across all channels to
+        trigger a detection
+    :param condition: Custom condition function which takes a dictionary with
+        correlations {seed_id: numpy.array} as argument and returns a boolean
+        array of the same length as the correlations.
+        Example condition function which sets a sample to True when at least
+        60% of the components have a correlation value larger than 0.8:
+
+        .. sourcecode:: python
+
+            def component_thresshold(ccs):
+                ccmatrix = np.array(list(ccs.values()))
+                num = np.count_nonzero(ccmatrix > 0.8, axis=0)
+                return num / len(ccs) > 0.6
+
+    :param template_time: UTCDateTime associated with template event
+        (e.g. origin time, default is the startime of the template stream)
+    :param holdon: Hold time in seconds, only the detection with largest
+        correlation value in that time span is returned.
+        (default: duration of template stream)
+    :param holdoff: Hold time in seconds, detections are
+        separated by this time span. (default: duration of template stream)
+    :param plot_ccmean: plot mean cross correlation with detections and
+        threshold
+    :param kwargs: kwargs are passed to correlate_template function
+
+    :return: list of UTCDateTime objects with detections and
+        dictionary with correlations {seed_id: numpy.array}
+    """
+    # preparations
+    stream.merge(fill_value=0)
+    if len({tr.stats.sampling_rate for tr in stream + template}) > 1:
+        raise ValueError('Traces have different sampling rate')
+    ids = {tr.id for tr in stream} | {tr.id for tr in template}
```

Comment:
I think you mean `ids = {tr.id for tr in stream} & {tr.id for tr in template}`

Reply:
Yea, you are right